### DISCUSSION TITLE
I would like to ask you a technical question: The maximum limit of tokens in openAi's official interface is 4096. I want to use the contextual dialogue function, but it easily exceeds the limit of tokens. How should I handle it better

### DISCUSSION THREAD

**Discussion by eamd-wq** [waylaidwanderer/node-chatgpt-api#322] [Discussions]
I would like to ask you a technical question: The maximum limit of tokens in openAi's official interface is 4096. I want to use the contextual dialogue function, but it easily exceeds the limit of tokens. How should I handle it better

**Comment by rocket-pig** [comment]
Well...leverage whatever language you are using to trim your input to the limit. Ask any of the AIs to help you and they'll straight up write the code for you! The more specific you are, the better the code is. I use javascript. Here's what they gave me:

_'This function callAPI takes in the new input as a string, adds it to the buffer, and then concatenates as many previous inputs from the buffer as possible while staying within the token limit. It then sends the concatenated string to the API and removes the processed inputs from the buffer.'_
```
const buffer = [];
const MAX_TOKENS = 4096;

function callAPI(input) {
  buffer.push(input);

  let prevInputs = "";
  let startIndex = 0;
  for (let i = buffer.length - 1; i >= 0; i--) {
    if (prevInputs.length + buffer[i].length <= MAX_TOKENS) {
      prevInputs = buffer[i] + " " + prevInputs;
      startIndex = i;
    } else {
      break;
    }
  }

  const query = encodeURIComponent(prevInputs.trim());
  const url = `https://example.api.com/?q=${query}`;

  fetch(url)
    .then(response => response.json())
    .then(data => console.log(data))
    .catch(error => console.error(error));

  buffer.splice(0, startIndex);
}

callAPI("new input");
```

Hope that's helpful!
